src/server/server.rs

```diff
@@ -549,12 +528,16 @@ impl<T: RaftStoreRouter, S: StoreAddrResolver> Handler for Server<T, S> {
         if events.is_writable() {
             self.on_writable(event_loop, token);
         }
+
+        if events.is_hup() {
```

Comment:
If it's readable, all the data has been read via `self.on_readable` before reaching this line.